packages/core/src/recipes.ts

```diff
@@ -73,9 +77,14 @@ export class Recipes {
       match,
       config: recipe,
       splitProps: (props) => {
-        return splitProps(props, variantKeys) as [Dict, Dict]
+        const [recipeProps, styleProps] = splitProps(props, variantKeys) as [Dict, Dict]
+        // omit other's recipe variant keys from the style props
+        return [recipeProps, omit(styleProps, Array.from(sharedState.variantKeys))]
```

Comment:
I’m not sure I understand the reasoning behind this. Feels like this might have some unintended effects